src/runtime/sgx/trusted/ecall_registry.cc

```diff
@@ -0,0 +1,82 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file ecall_registry.cc
+ * \brief The global registry of packed functions available via ecall_packed_func.
+ */
```

Comment:
consider change this to ecall_registry.h, and add def guard because the macro def and registry really should have sit in header file(even though it does not matter because of amalgamation)